ambry-tools/src/main/java/com.github.ambry/store/BlobStatus.java

```diff
@@ -61,22 +59,36 @@ void addAvailable(String replica) {
     return unavailable;
   }
 
-  void addDeletedOrExpired(String replica) {
+  boolean belongsToRecentIndexSegment() {
+    return belongsToRecentIndexSegment.get();
+  }
+
+  void setBelongsToRecentIndexSegment(boolean belongsToRecentIndexSegment) {
+    this.belongsToRecentIndexSegment.compareAndSet(false, belongsToRecentIndexSegment);
```

Comment:
could be just 
```
this. belongsToRecentIndexSegment = this. belongsToRecentIndexSegment || belongsToRecentIndexSegment
```
don't see why `compareAndSet()` (which is used in multithreading scenarios) is needed.